src/proto.rs

```diff
@@ -215,17 +215,18 @@ pub trait Codec {
 }
 
 impl<M: protobuf::Message> Codec for M {
-    type E = protobuf::error::ProtobufError;
+    type E = protobuf::Error;
 
     fn size(&self) -> u32 {
-        self.compute_size()
+        self.compute_size() as u32
     }
 
     fn encode(&self) -> Result<Vec<u8>, Self::E> {
         let mut buf = vec![0; self.compute_size() as usize];
         let mut s = CodedOutputStream::bytes(&mut buf);
         self.write_to(&mut s)?;
         s.flush()?;
+        drop(s);
```

Comment:
The `drop` is a fix for 
```
error[E0505]: cannot move out of `buf` because it is borrowed
   --> src/proto.rs:230:12
    |
226 |         let mut s = CodedOutputStream::bytes(&mut buf);
    |                                              -------- borrow of `buf` occurs here
...
230 |         Ok(buf)
    |            ^^^ move out of `buf` occurs here
231 |     }
    |     - borrow might be used here, when `s` is dropped and runs the `Drop` code for type `CodedOutputStream`

For more information about this error, try `rustc --explain E0505`.
error: could not compile `ttrpc` due to previous error
```

Reply:
ok, rust is not smart enough :)